server/plugin.go

```diff
@@ -92,11 +112,39 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
 		p.handleAccept(w, r)
 	case "/bump":
 		p.handleBump(w, r)
+	case "/telemetry":
+		p.handleTelemetry(w, r)
 	default:
 		http.NotFound(w, r)
 	}
 }
 
+type telemetryAPIRequest struct {
+	Event      string
+	Properties map[string]interface{}
+}
+
+func (p *Plugin) handleTelemetry(w http.ResponseWriter, r *http.Request) {
```

Comment:
IIRC, in the core product the client does the requests the telemetry platform itself. Is is something we should adopt for plugins?

If not, can we move this code into the library as well?

Reply:
My point is not sharing the key to the clients (my own concern, probably not a real concern) and to keep one single "source of truth" (the server). Also, it helps to keep some privacy for the clients, since other information (that might be useful or not) like carrier or so that probably is automatically fetched by rudder will not be fetched.

0/5 on whether or how this should be added to the library. We could add the whole handling like we are doing for the settings panel, or we can just add the handleTelemetry function and leave the routing to the plugin.

Reply:
Personally, I like it with the server doing the reporting and one less moving part for people to track and more consistent across plugins.

Reply:
Good points @larkox :+1: